gabarit/template_nlp/nlp_tests/functional_tests.py

```diff
@@ -31,17 +31,98 @@
+    x_test = x_train + x_valid + ['logistique technique', 'triage fabricants', 'fabricants fabricants']
+    model.fit(x_train=x_train, y_train=y_train, x_valid=x_valid, y_valid=y_valid)
+    model.save()
+    model_name = os.path.split(model.model_dir)[1]
+
+    dict_change_file = {'sklearn_pipeline_file': ('sklearn_pipeline_standalone.pkl', 'sklearn_pipeline_standalone_2.pkl'),
+                        'weights_file': ('best.hdf5', 'best_2.hdf5'),
+                        'tokenizer_file': ('embedding_tokenizer.pkl', 'embedding_tokenizer_2.pkl'),
+                        'tfidf_file': ('tfidf_standalone.pkl', 'tfidf_standalone_2.pkl'),
+                        'aggregation_function_file': ('aggregation_function.pkl', 'aggregation_function_2.pkl'), 
+                        'hf_model_dir': ('hf_model', 'hf_model_2'),
+                        'hf_tokenizer_dir': ('hf_tokenizer', 'hf_tokenizer_2')}
+
+    if change_file is None:
+        basic_run = f"{activate_venv}python {full_path_lib}/test_template_nlp-scripts/utils/0_reload_model.py -m {model_name}"
+    else:
+        old_file_name, new_file_name = dict_change_file[change_file]
+        shutil.move(os.path.join(model.model_dir, old_file_name), os.path.join(model.model_dir, new_file_name))
+        basic_run = f"{activate_venv}python {full_path_lib}/test_template_nlp-scripts/utils/0_reload_model.py -m {model_name} --{change_file} {new_file_name}"
+    test_class.assertEqual(subprocess.run(basic_run, shell=True).returncode, 0)
+
+    path_to_model = os.path.split(model.model_dir)[0]
+    new_model_name = get_last_model_created(path_to_model)
+    new_model_dir = os.path.join(path_to_model, new_model_name)
+    new_model, new_conf = model_class.ModelClass.load_model(model_dir=new_model_dir)
+    test_same_model_predictions(test_class, model, new_model, x_test)
+    test_class.assertNotEqual(model.model_dir, new_model.model_dir)
+    return model, new_model
+
+
+def test_same_model_tfidf(test_class, model, new_model, name_sub_model, model_equal_attributes, model_almost_equal_attributes):
+    tfidf = model.tfidf
+    new_tfidf = new_model.tfidf
+    for attribute in ['max_df', 'min_df']:
+        test_class.assertAlmostEqual(getattr(tfidf, attribute), getattr(new_tfidf, attribute))
+    for attribute in ['ngram_range', 'norm']:
+        test_class.assertEqual(getattr(tfidf, attribute), getattr(new_tfidf, attribute))
+    sub_model = getattr(model, name_sub_model)
+    new_sub_model = getattr(new_model, name_sub_model)
+    for attribute in model_equal_attributes:
+        test_class.assertEqual(getattr(sub_model, attribute), getattr(new_sub_model, attribute))
+    for attribute in model_almost_equal_attributes:
+        test_class.assertAlmostEqual(getattr(sub_model, attribute), getattr(new_sub_model, attribute))
+
+
+def test_same_model_not_tfidf(test_class, model, new_model, equal_attributes, almost_equal_attributes):
```

Comment:
The name of this function is a bit confusing and could be changed to "test_same_model"